views/default/forms/register.php

```diff
@@ -8,6 +8,11 @@
 
 if (elgg_is_sticky_form('register')) {
 	$values = elgg_get_sticky_values('register');
+
+	// Add the sticky values to $vars so views extending
+	// register/extend also get access to them.
+	$vars = array_merge($vars, $values);
+
```

Comment:
If it's safe to simply merge them, we might as well get rid of the separate $values array and just use $vars.